scripts/create_build_id.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+clean_tag_element() {
+    local tag_element="$1"
+    echo "${tag_element//\//-}"
+}
+
+generate_tag() {
+    local branch_name="$1"
+    local build_id="$2"
+    local git_hash="$3"
+
+    local clean_branch_name
+    local clean_build_id
+
+    clean_branch_name=$(clean_tag_element "$branch_name")
+    clean_build_id=$(clean_tag_element "$build_id")
+
```

Comment:
from my perspective it would make sense to declare hash as a separate variable before using it in a tag:
`local short_git_hash="${git_hash:0:7}"`
and then use it in tag:
`echo "${clean_branch_name}-${clean_build_id}-${short_git_hash}"`